### PR TITLE
associations merge to develop for the private-events project

### PR DESCRIPTION
**Pull request by eliasecasta**
In this project we created the sign up, sign in, events and users forms and views in order to display events created by certain user as well as attendees to the events. They are divided between past and future events.

### CODE REVIEW

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great start @team :clap: However, some requirements need to be implemented in order to get approval for this project. Check :arrow_down: 

#### Behaviour (Required) :warning: 

* [x] The event's show page displays the list of attendants ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/events_attendees_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
#### Implementation (required) :warning: 
##### UsersController

###### `show` action

* [x] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
#### Routes

* [x] routes only for used actions, i.e. `resources :events, only: [:new, :create, :show, :index]`
#### Views & Helpers

* [x] There should be NO application logic in the views nor helpers (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...). Everything connected with data processing should be encapsulated in models and used in controllers.
* [x] View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)
* [x] There should be absolutely NO logic in views.

  > Views are only for displaying information that became available to it from its corresponding controller not generating it. **Resources**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
#### Tests (required) :warning: 

* [ ] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- Some spec files are redundant, I think they should be merged.
### That's it :relieved: When finished, kindly submit a code re-review request :pray: 
Happy coding :v: 
**_Ahmed Mahfoudh_**

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  before_action :authenticated?
+  before_action :set_event, only: %i[show edit update destroy]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @future_events = Event.upcoming
+    @past_events = Event.past
+    @my_past = Event.my_past(current_user[:id])
+    @my_upcoming = Event.my_upcoming(current_user[:id])
```
Comment (Dandush03) [inline app/controllers/events_controller.rb]: - [ ] try using current_user.my_events.past

**Review by Dandush03 — CHANGES_REQUESTED** [review]
# status: :red_circle: Require Change :red_circle: 
### Great work so far
following my colleague request:

### Views & Helpers
- [ ] No logic in the views.
> [Check this out](http://codefol.io/posts/Where-Do-I-Put-My-Code)
### Routes
- [x] Be sure to use resources
> [Check out this docs for more insite](https://guides.rubyonrails.org/routing.html)
### Test
- [x] You have some failing tests

## Now let's dive in your code
# Happy Coding :computer: :nerd_face:

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (layzrd) [reply app/controllers/events_controller.rb]: we tried that but it doesn't work

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team
### Good job so far :clap: However, you have a few issues to solve:-
- [ ] Unit tests are not passing, rather they give out errors, solve this
- [ ] The are duplicated on display, views. Solve this. See the picture below
double
- [ ] Do a better design for the apps, they are clustered 
### Make the changes and submit for another code review. Good luck.
- Happy hacking!

By Raymond
Slack @iamraymond

**Review by christianotieno — APPROVED** [review]
# PROJECT APPROVED :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
***All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.***
- Congratulations on the splendid job :+1:
- You can now merge these changes.
- I wish you all the very best in your next tasks 🥇 
##
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)

**Pull request closed without merging**